aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/AbstractXRayInterceptor.java

```diff
@@ -0,0 +1,102 @@
+package com.amazonaws.xray.spring.aop;
+
+import com.amazonaws.xray.AWSXRay;
+import com.amazonaws.xray.entities.Subsegment;
+import com.amazonaws.xray.exceptions.SegmentNotFoundException;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Pointcut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class AbstractXRayInterceptor {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(this.getClass());
+
+    protected Object processXRayTrace(ProceedingJoinPoint pjp) throws Throwable {
+        boolean endSegment = false;
+        try {
+            if (AWSXRay.getCurrentSegment() != null) {
+                LOGGER.trace("Current segment exists");
+            }
+        } catch (SegmentNotFoundException snfe) {
+            LOGGER.trace("Creating new segment");
+            AWSXRay.beginSegment(pjp.getClass().getSimpleName());
+            endSegment = true;
+        }
+
+        try {
+            Subsegment subsegment = AWSXRay.beginSubsegment(pjp.getSignature().getName());
+            subsegment.setMetadata(this.generateMetadata(pjp, subsegment));
+            return XRayInterceptorUtils.conditionalProceed(pjp);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
```

Comment:
I think we should avoid re-logging any exceptions that may pass through customer code, as they likely will be logging it already.